main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopZoneIdValidator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.ImmutableMap;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.SchemaExport;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFareRuleTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsLocationType;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+
+/**
+ * Validates that all stops in "stops.txt" have a value for {@code stops.zone_id} if fare
+ * information is provided using "fare_rules.txt". This rule does not apply if a record from
+ * "stops.txt" represents a station or station entrance i.e {@code stops.location_type = 1 or 2}.
+ *
+ * <p>Generated notice: {@link StopWithoutZoneIdNotice}.
+ */
+@GtfsValidator
+public class StopZoneIdValidator extends FileValidator {
+
+  private final GtfsStopTableContainer stopTable;
+  private final GtfsFareRuleTableContainer fareRuleTable;
+
+  @Inject
+  StopZoneIdValidator(GtfsStopTableContainer stopTable, GtfsFareRuleTableContainer fareRuleTable) {
+    this.stopTable = stopTable;
+    this.fareRuleTable = fareRuleTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    if (fareRuleTable.getEntities().isEmpty()) {
+      return;
+    }
+    for (GtfsStop stop : stopTable.getEntities()) {
+      if (isStationOrEntrance(stop)) {
+        return;
```

Comment:
This `return` exits the whole `validate` function and interrupts validation of the rest of the file. Same for the `return` below.

Reply:
The logic here has been modified in #972 where `validate` is as follows:

https://github.com/MobilityData/gtfs-validator/blob/18c79215114536117b748e87c9c9b8288f068cc8/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopZoneIdValidator.java#L48